pkg/apis/core/types_shoot.go

```diff
@@ -984,6 +984,8 @@ type Worker struct {
 	Zones []string
 	// MachineControllerManagerSettings contains configurations for different worker-pools. Eg. MachineDrainTimeout, MachineHealthTimeout.
 	MachineControllerManagerSettings *MachineControllerManagerSettings
+	// Architecture is CPU architecture of machines in this worker pool.
+	Architecture *string
```

Comment:
According to the last discussions such field would make sense only if a machine type can support multiple architectures. Did you double-checked this?